### PR TITLE
Add option to have the search widget under item panel

### LINKED ISSUE #317: Search bar relocation request
I'm currently in the process of updating my original shanghaied with a number of mods I've discovered since publishing it and among them is a handy EZ storage mod that can be expanded to include a search bar and crafting table. It works really well with NEI or JEI for auto crafting the contents of the storage facility.

My problem however is that I always have the video on AUTO for ease of viewing which then covers the search bar of NEI/JEI. I've tried setting the video on NEI/JEI to old TMI as that program had the search bar on the bottom right of the screen but your 'old' TMI setting doesn't change the search bar location. Now I know the 1.7.10 version is not played much anymore but for me it was and still is the best version before all of the changes introduced in later versions.

So to this end is there any chance an option to change the location of your search bar could be introduced in an update?

Cheers.

### PR DESCRIPTION
**Pull request by Caedis**
Closes #317

New option (default true)
image

Center Search + Quantity Widget:
image

Center Search off + Quantity Widget:
image

Center Search off + Quantity Widget off:
image

Thoughts on increasing the size of the quantity widget to match the search widget?

### CODE REVIEW

**Comment by Caedis** [comment]
Also not sure if the other locale files need updating or if they default to the English locale if its not found.

**Review by wlhlm — CHANGES_REQUESTED** [review]
Nice change overall 👍 Tested and working nicely.

Yes, the translation will default to English, so everything's fine on that front.

I like the vertical arrangement of the +/- buttons. However I'd expect those buttons to be on the right of the input field having used similar widgets in other applications/toolkits. My main example would be the HTML5 number input field. Maybe this something I'm used to from left-to-right locales/languages.

Either way, I think the +/- need to be closer to the stack size input field instead of the search field. Currently it looks like they belong to the search field.

**Comment by Dream-Master** [comment]
@Caedis

File: src/main/java/codechicken/nei/ItemPanel.java [hunk 1]
```diff
@@ -166,19 +166,32 @@ public int getHeight(GuiContainer gui) {
...
+            more.y = less.y = quantity.y = y + h - BUTTON_SIZE;
+            less.x = x;
+            quantity.x = x + BUTTON_SIZE + 2;
+            quantity.w = more.x - quantity.x - 2;
+        } else {
+            more.x = less.x;
+            more.y = y + h - BUTTON_SIZE;
+            more.h = less.h = BUTTON_SIZE / 2;
+
+            less.x = (int) (x + (w * 0.7));
```
Comment (OneEyeMaker) [inline src/main/java/codechicken/nei/ItemPanel.java:187]: Should this coefficient be constant (0.7) or depend on `NEIClientConfig.showItemQuantityWidget()`?

**Review by OneEyeMaker — DISMISSED** [review]

**Comment by Caedis** [comment]
Gotcha, how does this look?
image

Still thinking if I want to increase the size of the search bar

**Comment by wlhlm** [comment]
> Gotcha, how does this look?

Exactly what I had in mind. 👍 

> Still thinking if I want to increase the height of the search bar

I agree. Maybe the same height as the centered search bar (and increase the stack size bar to be the same height)? Or parity with JEI? I hardly ever played with JEI, so I don't really remember how its layout is exactly...

**Comment by Caedis** [comment]
JEI does not have the quantity widget

**Comment by wlhlm** [comment]
I was thinking about the search bar height specifically.

**Comment by Caedis** [comment]
Oh, yeah

image

**Comment by Caedis** [comment]
and with no quantity widget (and less columns)
image

**Comment by Caedis** [comment]
Lemme see how it looks in GTNH then i'll commit

**Comment by wlhlm** [comment]
I think we should increase the quantity widget in the default layout as well to match the search bar layout. What do you think? Makes it more consistent.

**Comment by Caedis** [comment]
image

**Comment by wlhlm** [comment]
Nice! Waiting for your push.

**Review by wlhlm — APPROVED** [review]

**Pull request merged**